### DISCUSSION TITLE
How to Setup Audio Session with Audio Player?

### DISCUSSION THREAD

**Discussion by 8kdesign** [sbooth/SFBAudioEngine#278] [Q&A]
Hi! I'm trying to use the audio player in my app and I'm encountering a few problems that I suspect has to do with the audio session. Is there anything I need to do to get it working?

1. Resume/pause does not update MPNowPlayingInfoCenter, notification does not pause or resume
2. Audio is unable to play when device sleep, even when background mode for audio is enabled
3. If audio session is activated/deactivated manually, the audio player stops playing

Thank you

**Comment by sbooth** [comment]
I recently committed a fix for an issue with audio session interruptions that might remedy your third problem.

Support for `MPNowPlayingInfoCenter` isn't baked in to `SFBAudioPlayer` but I think you could get it working with the delegate method `audioPlayerPlaybackStateChanged` and then setting the playback state appropriately.
